[DX-1590] Tyk Operator : API Level Global Features (Draft)

dcs3spp opened the pull request:
### **User description**
### For internal users - Please add a Jira DX PR ticket to the subject!
[DX-1590](https://tyktech.atlassian.net/browse/DX-1590)

---

### Preview Link

[preview - link to internal looping from URLRewrite page](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/url-rewrite-tyk-classic)
[preview - internal looping](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-operator/advanced-configurations/internal-looping)

[preview - request header transformations](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/request-header-tyk-classic)
[preview - response header transformations](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/response-header-tyk-classic)
[preview - segment tags](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/advanced-configuration/manage-multiple-environments/with-tyk-multi-cloud)
[preview - CORS](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/tyk-apis/tyk-gateway-api/api-definition-objects/cors)
[preview - gRPC](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/plugins/supported-languages/rich-plugins/grpc/write-grpc-plugin/#tyk-operator)
[preview - Javascript](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-operator/advanced-configurations/custom-plugins)
[preview - Custom Plugin Analytics](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/plugins/plugin-types/analytics-plugins/)
[preview - Round Robin Load Balancing](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/planning-for-production/ensure-high-availability/load-balancing/)

[preview - Do not track middleware](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/do-not-track-tyk-classic)
[preview - Analytics tagging](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/tyk-apis/tyk-gateway-api/api-definition-objects/custom-analytics/)

### Description

Add examples for API level global features. The PR includes all examples except for Config Data which had target page marked as missing. The config data example is included in virtual endpoints.

Question for reviewers:

- Is there a working example for Tyk Operator for configuring gRPC to use remote bundles? If plugin bundles are not supported in Tyk Operator then content in PR should be updated to state this. There is an example in this PR for gRPC with custom middleware manifest in API definition but not an example that links to the bundled zip file containing the manifest.

- Should the following API global features be included with examples?
    - [Analytics API tagging](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/tyk-apis/tyk-gateway-api/api-definition-objects/custom-analytics/) - **_Done example added ready for review_**
    - [Do not track analytics (per API)](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/do-not-track-tyk-classic) - **_Done example added ready for review_**
    - Custom plugins:
        - Go
        - Lua
        - Python
       
#### Screenshots (if appropriate)
<br>

## Checklist
<!-- Go over all the following points, and put an `x` in all the boxes that apply -->

- [x] I have added a **preview link** to the PR description.
- [x] I have reviewed the suggestions made by our AI (PR Agent) and updated them accordingly (spelling errors, rephrasing, etc.)
- [x] I have [reviewed the guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING.md) for contributing to this repository.
- [x] I have [read the technical guidelines](https://github.com/TykTechnologies/tyk-docs/blob/master/CONTRIBUTING-TECHNICAL-GUIDE.md) for contributing to this repository.
- [x] Make sure you have started *your change* off *our latest `master`*.
- [x] I **labeled** the PR
<!-- Label your PR according to the type of changes that your code introduces. This ensures that we know how/when to publish the PR. These are the options:
- Fixing typo (please merge to production) - add the label `now`
- Documenting a new feature (please merge to production) - add the label `now`
- Documentation for future release (please do not merge to production) - add label `future release` and label of the release
- [Something else (please add if needs merging to production or not) - add label according to the above logic -->


[DX-1590]: https://tyktech.atlassian.net/browse/DX-1590?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ


___

### **PR Type**
Documentation


___

### **Description**
- Added extensive documentation for Tyk Operator, including examples for API shard tagging, global rate limiting, load balancing, and CORS configuration.
- Introduced new sections on configuring custom plugins and internal looping with Tyk Operator, providing detailed examples and YAML configurations.
- Enhanced existing documentation with Tyk Operator examples for analytics plugins, gRPC plugin hooks, request and response header transformations, and middleware configuration.
- Updated menu and reference links to include new documentation entries and ensure consistency.


___



### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Documentation</strong></td><td><details><summary>13 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>with-tyk-multi-cloud.md</strong><dd><code>Add Tyk Operator API Shard Tagging Example</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/advanced-configuration/manage-multiple-environments/with-tyk-multi-cloud.md

<li>Added a section on tagging an API for a shard using Tyk Operator.<br> <li> Provided a YAML example for API definition with tags.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-19a2c34107181ccee3dc99cb4e639868ae543027dee22aa994a78fb356e36913">+23/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>rate-limiting.md</strong><dd><code>Add Tyk Operator Global Rate Limiting Example</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/basic-config-and-security/control-limit-traffic/rate-limiting.md

<li>Added an example of using Tyk Operator for global rate limiting.<br> <li> Provided a YAML configuration for setting rate limits.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-1d63c4783ce4de0d1fbaed0b2a6e1ed70cf4a5c0557f1790439e294463127572">+22/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>load-balancing.md</strong><dd><code>Configure Load Balancing with Tyk Operator</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/planning-for-production/ensure-high-availability/load-balancing.md

<li>Introduced configuration for load balancing via Tyk Operator.<br> <li> Included a YAML example for enabling load balancing.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-cbad277a7f953c4004b2976f975c9c41ead538c43910e5b227fb9b6d6e362677">+30/-1</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>analytics-plugins.md</strong><dd><code>Add Tyk Operator Analytics Plugin Configuration</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/plugins/plugin-types/analytics-plugins.md

<li>Added Tyk Operator example for configuring analytics plugins.<br> <li> Provided YAML configuration for enabling analytics plugins.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-a1ac3479e138cc32d45789e39591e91d2983c713bca94825ab56fe6262b1c906">+49/-2</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>write-grpc-plugin.md</strong><dd><code>Configure gRPC Plugin Hooks with Tyk Operator</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/plugins/supported-languages/rich-plugins/grpc/write-grpc-plugin.md

<li>Added examples for configuring gRPC plugin hooks using Tyk Operator.<br> <li> Included pre and post plugin hook examples.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-efbf2d8054bc1cc8ce107b77a57ae5185f3f41accbc5b8a7a4545c599dde305d">+53/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>do-not-track-tyk-classic.md</strong><dd><code>Enhance Middleware Configuration for Tyk Operator</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-gateway/middleware/do-not-track-tyk-classic.md

<li>Expanded section on configuring middleware in Tyk Operator.<br> <li> Added examples for disabling tracking for all or selected endpoints.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-f3d63cffdb9b07b7b4a8291c3777be7013fc890c94971cd6c93ec156c58d8524">+29/-4</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>request-header-tyk-classic.md</strong><dd><code>Add Request Header Transformation Example for Tyk Operator</code></dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-gateway/middleware/request-header-tyk-classic.md

<li>Added example for API level request header transformations using Tyk <br>Operator.<br> <li> Provided YAML configuration for adding and removing headers.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-2b8d7eb6abcd81e52f5955ba294009309b94ab4db7a34d53274266d94340b985">+39/-2</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>url-rewrite-tyk-classic.md</strong><dd><code>Reference Internal Looping Examples</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-gateway/middleware/url-rewrite-tyk-classic.md

- Added reference to internal looping examples.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-3da33c570cc1f0bc531d763c8671f21b97f6a366a9670fe89d17da982fc63838">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>custom-plugins.md</strong><dd><code>Document Custom Plugin Configuration with Tyk Operator</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-operator/advanced-configurations/custom-plugins.md

<li>Introduced documentation for configuring custom plugins using Tyk <br>Operator.<br> <li> Provided a detailed example for JavaScript plugin configuration.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-d69628c1fce8c472372ca00cae0a07ab58dd913b909fe191fb1732c557c2f8a9">+88/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>internal-looping.md</strong><dd><code>Document Internal Looping with Tyk Operator</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-operator/advanced-configurations/internal-looping.md

<li>Added comprehensive documentation on internal looping with Tyk <br>Operator.<br> <li> Included examples for URL rewrites, triggers, and proxy <br>configurations.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-d29f5dc16772912c912da12c53a4a6e2bcf2858bd476e55ce77a7bd569783368">+344/-0</a>&nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>api-definition.md</strong><dd><code>Update JavaScript Plugin Reference Link</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-operator/reference/api-definition.md

- Updated reference link for JavaScript custom plugins.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-2a3c59b9948a8a992bebbaecc84925ccbe04564820a47e3274cd189823ddea14">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>cors.md</strong><dd><code>Add CORS Configuration Example for Tyk Operator</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-apis/tyk-gateway-api/api-definition-objects/cors.md

<li>Added examples for enabling CORS in Tyk Operator.<br> <li> Provided YAML configuration for CORS settings.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-a5bdb402df3f2fd991f22cef4b99746171ea81af4ba86f79a5c61d1ba1c8fd14">+43/-3</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>custom-analytics.md</strong><dd><code>Setup Tag Headers with Tyk Operator</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/tyk-apis/tyk-gateway-api/api-definition-objects/custom-analytics.md

<li>Added section on setting up tag headers with Tyk Operator.<br> <li> Included YAML example for tagging headers.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-23a35627af11d924b7a5b88273039d75890ee084e1ebd83f9196a2cb91f1ba0e">+28/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Formatting</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>response-header-tyk-classic.md</strong><dd><code>Minor Formatting Adjustment</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/content/product-stack/tyk-gateway/middleware/response-header-tyk-classic.md

- Minor formatting adjustment.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-7ad752916be249ca461116af1160a2b74251e94b08402e4ab3f04db7c41fe638">+1/-1</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr><tr><td><strong>Configuration changes</strong></td><td><details><summary>1 files</summary><table>
<tr>
  <td>
    <details>
      <summary><strong>menu.yaml</strong><dd><code>Update Menu with New Documentation Entries</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

tyk-docs/data/menu.yaml

- Added menu entries for Custom Plugins and Internal Looping.


</details>


  </td>
  <td><a href="https://github.com/TykTechnologies/tyk-docs/pull/5175/files#diff-a6789c66bfa0660f7fdbe4796002ab4777677bae59a0276689c840100ff0b0b2">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></details></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

dcs3spp commented:
closing until preview fixed, preview not being generated for PR, previously it was working

letzya commented on tyk-docs/content/product-stack/tyk-operator/advanced-configurations/internal-looping.md:
```diff
@@ -0,0 +1,344 @@
+---
```
Can I please ask why we have a new file for this capability and not updating an existing doc?

dcs3spp replied:
@letzya In spreadsheet provided by @caroltyk the target docs location was given as _product-stack/tyk-operator/advanced-configurations/internal-looping.md_ within Tyk Operator docs

dcs3spp replied:
@caroltyk what was the reason for targeting the [internal looping example](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-operator/advanced-configurations/internal-looping) within Tyk Operators docs?

Are you ok with that content being moved to this [page](https://tyk.io/docs/advanced-configuration/transform-traffic/looping/#advanced-configuration) as an additional section? 

letzya replied:
@dcs3spp looks like this is a looping page that doesn't mention looping https://tyk.io/docs/product-stack/tyk-gateway/middleware/internal-endpoint-middleware/ - maybe we want to add the content here?

caroltyk replied:
We can move the content to https://tyk.io/docs/product-stack/tyk-gateway/middleware/internal-endpoint-middleware/ 
To align the terminologies let's update the "Looping" to "Internal Endpoints" in the reference table too.

caroltyk replied:
Hi @dcs3spp Since operator has special feature to allow user to reference rewrite URL using k8s name and namespace, we can keep this page where it is. We can add a link from https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/url-rewrite-tyk-classic/ to Operator's looping page as cross reference.

dcs3spp replied:
@caroltyk I have updated with link from URL-Rewrite. Should there be a link updated from the Tyk Classic  page of the section highlighted by @letzya  https://tyk.io/docs/product-stack/tyk-gateway/middleware/internal-endpoint-tyk-classic/ ?

caroltyk replied:
I don't think looping example should be added there because it doesn't do looping, just blocking external access.

Btw, we have Internal Endpoint example on another PR too: https://deploy-preview-5258--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/internal-endpoint-tyk-classic/#tyk-operator

dcs3spp commented on tyk-docs/content/product-stack/tyk-gateway/middleware/request-header-tyk-classic.md:
```diff
@@ -137,4 +137,47 @@ Select the headers to delete and insert using the provided fields. You need to c
+
+The example below shows an `ApiDefinition` custom resource that adds *foo-req* and *bar-req* headers to the request before it is sent upstream. The *foo-req* header has a value of *foo-val* and the *bar-req* header has a value of *bar-val*. Furthermore, the *hello* header is removed from the request before it is sent upstream.
+
+<!-- QUESTION: Is this example valid? -->
```
Thanks @buraksekili I have removed the comment

caroltyk commented on tyk-docs/content/plugins/supported-languages/rich-plugins/grpc/write-grpc-plugin.md:
```diff
@@ -151,6 +155,55 @@ For configurations directly embedded within the Tyk Gateway, plugin hooks can be
 
+#### Tyk Operator
+
+The examples below illustrate how to configure configure plugin hooks for an API Definition within Tyk Operator:
```
```suggestion
The examples below illustrate how to configure plugin hooks for an API Definition within Tyk Operator:
```

caroltyk commented on tyk-docs/content/product-stack/tyk-gateway/middleware/url-rewrite-tyk-classic.md:
```diff
@@ -15,6 +15,8 @@ If you want to use dynamic data from context variables, you must [enable]({{< re
 
 If you're using the newer Tyk OAS APIs, then check out [this]({{< ref "/product-stack/tyk-gateway/middleware/url-rewrite-tyk-oas" >}}) page.
 
+If you're using Tyk Operator, then check out [this]({{< ref "/product-stack/tyk-operator/advanced-configurations/internal-looping" >}}) page.
```
Suggest to create a subheading at the end of the page for Tyk Operator and add the link there, so that users easily know they can configure this via Operator from the navigation menu.

dcs3spp replied:
Thanks @caroltyk I have updated to included subheading for Tyk Operator that includes the link

caroltyk commented on tyk-docs/content/product-stack/tyk-operator/advanced-configurations/internal-looping.md:
```diff
@@ -0,0 +1,344 @@
+
+The `rewrite_to_internal` object is used to define how incoming API requests should be redirected to internal API resources. This mechanism is essential for routing traffic within a microservices architecture or when managing APIs across different namespaces in Kubernetes. Using this object you can effectively manage and optimize API traffic within your Tyk Gateway.
+
+This setup supports complex routing scenarios and enables efficient inter-service communication, via configuring the following properties:
```
```suggestion
This setup supports complex routing scenarios and enables efficient inter-service communication, via configuring the following properties under `url_rewrites`:
```

caroltyk commented on tyk-docs/content/product-stack/tyk-operator/advanced-configurations/internal-looping.md:
```diff
@@ -0,0 +1,344 @@
+
+---
+
+## Configuration
```
```suggestion
## Configure URL rewrites to internal ApiDefinition resources
```

dcs3spp commented:
Thanks @caroltyk I have added example for [do not track middleware](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/do-not-track-tyk-classic). Should we add an example to that section also for setting do not track middleware for a specific endpoint?

caroltyk commented:
> Thanks @caroltyk I have added example for [do not track middleware](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/do-not-track-tyk-classic). Should we add an example to that section also for setting do not track middleware for a specific endpoint?

Do not track middleware example is added here: https://deploy-preview-5258--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/do-not-track-tyk-classic/#tyk-operator

dcs3spp commented:
> > Thanks @caroltyk I have added example for [do not track middleware](https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/do-not-track-tyk-classic). Should we add an example to that section also for setting do not track middleware for a specific endpoint?
> 
> Do not track middleware example is added here: https://deploy-preview-5258--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/do-not-track-tyk-classic/#tyk-operator

Thanks @caroltyk I have added the example to https://deploy-preview-5175--tyk-docs.netlify.app/docs/nightly/product-stack/tyk-gateway/middleware/do-not-track-tyk-classic
